test/oxo.js

```diff
@@ -0,0 +1,345 @@
+const Reverter = require('./helpers/reverter');
+const Asserts = require('./helpers/asserts');
+const OXO = artifacts.require('./OxoGame.sol');
+
+contract('OXO', function (accounts) {
+    const reverter = new Reverter(web3);
+    afterEach('revert', reverter.revert);
+
+    const asserts = Asserts(assert);
+    const OWNER = accounts[0];
+    let oxo;
+    let defaultBet = 1000;
+    let defaultGameId = 0;
+
+    before('setup', () => {
+        return OXO.deployed()
+            .then(instance => oxo = instance)
+            .then(reverter.snapshot);
+    });
+
+
+    it('should allow to create the game ', async () => {
+
+        const player = accounts[1];
+        let gameId = await oxo.createGame.call({from: player, value: defaultBet}).valueOf();
+        assert.equal(gameId, 0);
+    });
+
+    it('should deny to create the game with incorrect bet', async () => {
+
+        const player = accounts[1];
+        var res = await asserts.throws(oxo.createGame({from: player, value: 0}));
+
+        const value = '0xffffffsffffffffffffffffffffffffffffffffffffffffffffffffffffffffff';
+        var res = await asserts.throws(oxo.createGame({from: player, value: value}));
+
+
+        await asserts.throws(oxo.createGame({from: player, value: value / 2}));
+    });
+
+    it('should emit event about game creation ', async () => {
+
+        const player = accounts[1];
+        var result = await oxo.createGame({from: player, value: defaultBet});
+        assert.equal(result.logs[0].event, 'GameCreated');
+        assert.equal(result.logs[0].args.gameId.valueOf(), defaultGameId);
+        assert.equal(result.logs[0].args.bet.valueOf(), defaultBet);
+    });
+
+    it('should allow user to join the game', async () => {
```

Comment:
This test checks 'should emit event after join a game'. 
How else you can check that it's possible to join game?

Reply:
1) i check some cases when user cannot join to the game 
2) i check whether i emit event once player has joined.. is not it enough ?
3) if there are no 2 players  - nobody can do a move. It tests further tests :)

Reply:
The name of the test and its contents do not match. 
You could check isPlayer in some test or game status or maxPlayersCount etc. 
It is never enough tests until they cover all the cases))

Reply:
1)  yep, could be.. i just made all these function as internal and did not want to make then  visible to outside. But actually isPlayer and gameStatus could be accessible
2) maxPlayersCount is a constant and equals 2. That is why u maybe did not understand the purpose of line `(game.playerCount == maxPlayerCount)`